dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureIntrospector.java

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.kernel.feature.internal;
+
+import java.io.PrintWriter;
+import java.util.Set;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import com.ibm.ws.kernel.feature.FeatureDefinition;
+import com.ibm.ws.kernel.feature.FeatureProvisioner;
+import com.ibm.wsspi.logging.Introspector;
+
+/**
+ * Introspector implementation that prints server feature metadata
+ * in the event of a dump
+ */
+
+@Component
+public class FeatureIntrospector implements Introspector {
+
+    private final FeatureProvisioner provisioner;
+
+    @Activate
+    public FeatureIntrospector(@Reference FeatureProvisioner provisioner) {
+        this.provisioner = provisioner;
+    }
+
+    @Override
+    public String getIntrospectorName() {
+        return "FeatureIntrospector";
+    }
+
+    @Override
+    public String getIntrospectorDescription() {
+        return "Lists all the features enabled in the server.";
+    }
+
+    /**
+     * Requests the set of features that are installed on the server and outputs
+     * relevant data to the log file
+     */
+    @Override
+    public void introspect(PrintWriter out) throws Exception {
+        Set<String> features = this.provisioner.getInstalledFeatures();
+        for (String feature : features) {
+            FeatureDefinition featureDef = this.provisioner.getFeatureDefinition(feature);
+            out.println(featureDef.getFeatureName() + "[");
+            out.println("\tvisibility: " + featureDef.getVisibility());
+            out.println("\tversion: " + featureDef.getVersion());
+            out.println("\tsimple name: " + featureDef.getFeatureName());
+            out.println("\tsymbolic name: " + featureDef.getSymbolicName());
+            out.println("]");
```

Comment:
We should be able to safely cast the `featureDef` to `com.ibm.ws.kernel.feature.provisioning.ProvisioningFeatureDefinition` here.  I suggest we do an `instanceof` check and then cast and also provide the following information:

ProvisioningFeatureDefinition.getBundleRepositoryType()
ProvisioningFeatureDefinition.getConstituents(SubsystemContentType)
ProvisioningFeatureDefinition.getSupersededBy()
ProvisioningFeatureDefinition.isAutoFeature()
ProvisioningFeatureDefinition.isSingleton()